plugins/bufferlines/barbar.nix

```diff
@@ -69,249 +24,566 @@ let
     goTo7 = "Goto 7";
     goTo8 = "Goto 8";
     goTo9 = "Goto 9";
+    first = "First";
     last = "Last";
     pin = "Pin";
+    restore = "Restore";
     close = "Close";
+    closeAllButCurrent = "CloseAllButCurrent";
+    closeAllButVisible = "CloseAllButVisible";
+    closeAllButPinned = "CloseAllButPinned";
+    closeAllButCurrentOrPinned = "CloseAllButCurrentOrPinned";
+    closeBuffersLeft = "CloseBuffersLeft";
+    closeBuffersRight = "CloseBuffersRight";
+    wipeout = "Wipeout";
     pick = "Pick";
+    pickDelete = "PickDelete";
     orderByBufferNumber = "OrderByBufferNumber";
+    orderByName = "OrderByName";
     orderByDirectory = "OrderByDirectory";
     orderByLanguage = "OrderByLanguage";
     orderByWindowNumber = "OrderByWindowNumber";
   };
 in
-{
-  options.plugins.barbar = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "barbar.nvim";
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "barbar";
+  originalName = "barbar.nvim";
+  defaultPackage = pkgs.vimPlugins.barbar-nvim;
+  extraPlugins = [ pkgs.vimPlugins.nvim-web-devicons ];
```

Comment:
How do you decide if this dependency should be included? Upstream recommendation? (Should I do the same in the `dashboard` PR?)

Reply:
> How do you decide if this dependency should be included? Upstream recommendation? (Should I do the same in the `dashboard` PR?)

Well, this plugin is not mandatory per say. However, I feel like making an option to opt-out from this quite overkilled.
Also, this was the behavior for our `barbar` module prior to this refactor. Hence, I kept the same approach.